modules/google-speech/src/backend/middleware.ts

```diff
@@ -0,0 +1,136 @@
+import axios from 'axios'
+import * as sdk from 'botpress/sdk'
+import FormData from 'form-data'
+import { pick } from 'lodash'
+import { v4 as uuidv4 } from 'uuid'
+
+import { GoogleSpeechClient } from './client'
+import { Clients } from './typings'
+
+const INCOMING_MIDDLEWARE_NAME = 'googleSpeech.speechToText'
+const OUTGOING_MIDDLEWARE_NAME = 'googleSpeech.textToSpeech'
+
+export class Middleware {
+  constructor(private bp: typeof sdk, private clients: Clients) {}
+
+  public setup() {
+    this.bp.events.registerMiddleware({
+      description: 'Converts audio content to text using google speech-to-text.',
+      direction: 'incoming',
+      handler: this.incomingHandler.bind(this),
+      name: INCOMING_MIDDLEWARE_NAME,
+      order: 1
+    })
+
+    this.bp.events.registerMiddleware({
+      description: 'Converts text to audio using google text-to-speech.',
+      direction: 'outgoing',
+      handler: this.outgoingHandler.bind(this),
+      name: OUTGOING_MIDDLEWARE_NAME,
+      order: 1
+    })
+  }
+
+  private async incomingHandler(event: sdk.IO.IncomingEvent, next: sdk.IO.MiddlewareNextCallback) {
+    // TODO: Add more validation than the payload type being audio
+    if (event.payload.type !== 'audio') {
+      return next()
```

Comment:
The `next()` method has multiple parameters which can provide additional information when debugging on the emulator. In this condition, you skip processing the middleware. The third parameter of the next method is "skipped", so it would be nice to return `next(undefined, false, true)` when your MW doesn't do anything

![image](https://user-images.githubusercontent.com/42552874/116549649-c01ac100-a8c3-11eb-9061-4c6459968045.png)